core/src/main/scala/chisel3/internal/firrtl/IR.scala

```diff
@@ -861,7 +862,28 @@ case class DefBlackBox(
   params: Map[String, Param])
     extends Component
 
-case class Circuit(name: String, components: Seq[Component], annotations: Seq[ChiselAnnotation], renames: RenameMap) {
-  def firrtlAnnotations: Iterable[Annotation] = annotations.flatMap(_.toFirrtl.update(renames))
+@data class Circuit(
+  name:       String,
+  components: Seq[Component],
+  @deprecated("Do not use annotations val of Circuit directly - use firrtlAnnotations instead. Will be removed in a future release",
+    "Chisel 3.5")
+  annotations: Seq[ChiselAnnotation],
```

Comment:
@jackkoenig is this the right way to call deprecated? Do I need to change the version number - since I am gettnig a compile error (due to this deprecation) within our internal sifive codebase. I distinctly remember using Chisel 3.5 for something similar in the recent past and it not complaining.

Reply:
Added nowarn and now compile passes.